src/engine/runtime.js

```diff
@@ -1012,6 +1013,51 @@ class Runtime extends EventEmitter {
             this._step();
         }, interval);
     }
+
+    // ---------------------------------------------------------------------
+
+    /**
+     * Create event listener for variables.
+     * @param {object} e Blockly "var" event
+     */
+    blocklyListen (e) {
+        // Validate event
+        if (typeof e !== 'object') return;
+        if (typeof e.varId !== 'string') return;
+        const stage = this.getTargetForStage();
+        // Var create/rename/destroy
+        switch (e.type) {
+        case 'var_create':
+            {
+                if (!stage.variables.hasOwnProperty(e.varId)) {
+                    const newVariable = new Variable(e.varId, e.varName, 0,
+                        false);
+                    stage.variables[e.varId] = newVariable;
+                }
+                break;
+            }
+        case 'var_rename':
+            {
+                if (stage.variables.hasOwnProperty(e.varId)) {
+                    const variable = stage.variables[e.varId];
+                    if (variable.id === e.varId) {
+                        variable.name = e.newName;
+                    }
+                }
+                break;
+            }
+        case 'var_delete':
+            {
+                const id = e.varId;
+                if (stage.variables.hasOwnProperty(id)) {
+                    delete stage.variables[id];
+                }
+                break;
+            }
+        }
+    }
+    // ---------------------------------------------------------------------
```

Comment:
I am wondering about this listener being in Runtime. I think it might be better placed in the main [`Blocks#blocklyListen`](https://github.com/LLK/scratch-vm/blob/b9aefa780f40f2bdc1c44dca34683aadf044cc6e/src/engine/blocks.js#L190), for a few reasons:
* The VM may not be running in a context with scratch-blocks, so it would be nice to keep the scratch-blocks-specific code contained. I think it would help us later on if we avoided spreading it into multiple parts of the VM. Eventually we may refactor all of it out of the VM and into an adapter, so having it self-contained would make that easier to do.
* I believe it is possible to keep this contained in the main blocks listener: Although `optRuntime` is technically optional, in practice it's always available for the events coming from scratch-blocks. So we should have access to get the stage target (or access other methods on the runtime if we reorganize variables later).
* Currently all variables are global, so we are keeping the list of variables on the stage only. When we have scoped variables, we will want to store them by target. The current setup of piping events from scratch-blocks to the current editing target's block container would help with that, since if the variable is scoped, then we would attach it to whatever the listener's target is, otherwise the stage.

/cc @paulkaplan @fsih @cwillisf @thisandagain